src/search/lp/lp_solver.h

```diff
@@ -73,6 +78,31 @@ struct LPVariable {
                double objective_coefficient);
 };
 
+class LinearProgram {
+    LPObjectiveSense sense;
+    std::string objective_name;
+
+    utils::NamedVector<LPVariable> variables;
+    utils::NamedVector<LPConstraint> constraints;
+    double infinity;
+
+public:
+    explicit LinearProgram(LPObjectiveSense sense, utils::NamedVector<LPVariable> &&variables, utils::NamedVector<LPConstraint> &&constraints, std::string objective_name = "")
+        : sense(sense), objective_name(objective_name), variables(std::move(variables)), constraints(std::move(constraints)) {
```

Comment:
No, the objective name is a name you can pick for the linear expression in the objective function. For example "total_cost" in operator-counting LPs or "initial_h_value"/"average_h_value" for a potential heuristic LP. Although, now that I see this line again, maybe it would be more consistent with the names of variables and constraints if we use `set_objective_name()` rather than passing the value in the constructor.

Reply:
We set the names of variables and constraints upon initializing the lp_solver as well by passing in the NamedVectors at that point. I do suppose the variable and constraint names can be changed after initialization and the objective name currently cannot.

Reply:
In any case, I think the `objective_name` warrants some documentation along the lines of what Florian wrote.

Reply:
I meant inside of `NamedVector` we do not pass the name as an optional parameter to `push_back` anymore but switched to a method where we use `set_name` after we add the actual content. I think it would make sense to do the same here. It also makes it easier to implement things in a way that names are only used in some cases.